gcalcli/gcalcli.py

```diff
@@ -975,49 +978,53 @@ def _edit_event(self, event):
                 sys.exit(0)
 
             elif val.lower() == 't':
-                self.printer.msg('Title: ', 'magenta')
-                val = input()
+                val = get_input(self.printer, 'Title: ', STR_NOT_EMPTY)
                 if val.strip():
                     event['summary'] = val.strip()
 
             elif val.lower() == 'l':
-                self.printer.msg('Location: ', 'magenta')
-                val = input()
+                val = get_input(self.printer, 'Location: ', STR_ALLOW_EMPTY)
                 if val.strip():
                     event['location'] = val.strip()
 
             elif val.lower() == 'w':
-                self.printer.msg('When: ', 'magenta')
-                val = input().strip()
+                val = get_input(self.printer, 'When: ', PARSABLE_DATE).strip()
                 if val:
                     td = (event['e'] - event['s'])
                     length = ((td.days * 1440) + (td.seconds / 60))
+                    try:
+                        all_day = self.options.get('allday')
+                    except KeyError:
+                        all_day = None
                     try:
                         newStart, newEnd = utils.get_times_from_duration(
-                                val, length, self.options['allday'])
+                                val, length, all_day)
                     except ValueError as exc:
                         self.printer.err_msg(str(exc))
                         sys.exit(1)
                     event = self._SetEventStartEnd(newStart, newEnd, event)
 
             elif val.lower() == 'g':
-                self.printer.msg('Length (mins): ', 'magenta')
-                val = input().strip()
+                val = get_input(
+                    self.printer, 'Length (mins): ', STR_TO_INT)
                 if val:
+                    try:
+                        all_day = self.options.get('allday')
+                    except KeyError:
+                        all_day = None
                     try:
                         newStart, newEnd = utils.get_times_from_duration(
                                 event['start']['dateTime'], val,
-                                self.options['allday'])
+                                all_day)
```

Comment:
It looks to me like we don't actually do anything with newStart, newEnd here.  The convention in python for an unused variable is `_`, but here we have two!  I guess we can just not assign the return value, this is Python, after all.  So, does the 'g' key actually do anything?  This looks like some sort of bug to me (not your fault, and existing one).